### ISSUE TITLE
onSelected event in PlutoRow, PlutoCell and PlutoColumn

### ISSUE THREAD

**Issue by JanRegent** [bosskmk/pluto_grid#37]
It's possible to add onSelected event in PlutoRow, PlutoCell and PlutoColumn?

Thanks a lot for your work:-)

**Comment by bosskmk** [comment]
Do you want to receive an event every time the currently selected cell (and Row, Column) changes?

If so, you can achieve it with the example code below.

If this is not the case, please reply again.

```dart
import 'package:flutter/material.dart';
import 'package:pluto_grid/pluto_grid.dart';

import '../../dummy_data/development.dart';
import '../../widget/pluto_example_button.dart';
import '../../widget/pluto_example_screen.dart';

class MovingScreen extends StatefulWidget {
  static const routeName = 'feature/moving';

  @override
  _MovingScreenState createState() => _MovingScreenState();
}

class _MovingScreenState extends State<MovingScreen> {
  List<PlutoColumn> columns;

  List<PlutoRow> rows;

  PlutoStateManager stateManager;

  @override
  void dispose() {
    super.dispose();

    stateManager.removeListener(handleCurrentSelectionState);
  }

  @override
  void initState() {
    super.initState();

    final dummyDate = DummyData(10, 100);

    columns = dummyDate.columns;

    rows = dummyDate.rows;
  }

  void handleCurrentSelectionState() {
    print(stateManager.currentCell.value);
    print(stateManager.currentRow.key);
    print(stateManager.currentColumn.title);
  }

  @override
  Widget build(BuildContext context) {
    return PlutoExampleScreen(
      title: 'Moving',
      topTitle: 'Moving',
      topContents: [
        Text(
            'Change the current cell position with the arrow keys, enter key, and tab key.'),
      ],
      topButtons: [
        PlutoExampleButton(
          url:
              'https://github.com/bosskmk/pluto_grid/blob/master/example/lib/screen/feature/moving_screen.dart',
        ),
      ],
      body: PlutoGrid(
        columns: columns,
        rows: rows,
        onChanged: (PlutoOnChangedEvent event) {
          print(event);
        },
        onLoaded: (PlutoOnLoadedEvent event) {
          stateManager = event.stateManager;
          stateManager.addListener(handleCurrentSelectionState);
        },
      ),
    );
  }
}
```

**Comment by bosskmk** [comment]
f you want to add it as a feature, I will.
I'll add it with onCurrentSelected or whatever name you suggest.

**Comment by JanRegent** [comment]
Hi 
that's enough, thanks :)

 void handleCurrentSelectionState() {
  ...
  print(stateManager.currentRow.cells.values.first.value); //in first column is Google sheet row number
  }

I'm working on an application to easily filter data from google sheets

onCurrentSelected can be nice when you have time

**Comment by wwyung** [comment]
Sorry to add a followup question here even though the ticket has been closed. I found register a callback via addListener() not quite working the way I wanted. My pluto grid is situated near the bottom of my phone screen, when I edit a cell, the popup keyboard will block the view, hence I want to scroll the grid into view upwards only once when I start editing the cell, however I found the callback was hit multiple times (3 times to be exact) when I tap a cell to edit. Following code is what's in my callback:

  void handleCheckedRow() {
    if (_maintenanceHistStateMgr.isEditing) {
      Scrollable.ensureVisible(_maintenanceHistkey.currentContext!);   <== this is the line that got hit 3 times of the same cell
    }
  }

What I noticed is that only the 3rd call to the callback, the grid will briefly move upwards momentary after I finished editing. Not sure why the grid didn't move the first time. BTW, _maintenanceHistkey is the key to the grid parent.
